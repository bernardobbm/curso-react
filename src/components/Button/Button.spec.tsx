import { describe, expect, it, vitest } from 'vitest';
import { fireEvent, render, screen } from '@testing-library/react';
import { Button } from '.';

const fn = vitest.fn();

describe('Button Component', () => {
	it('should render the button with the text', () => {
		render(
			<Button
				onClick={fn}
				disabled={false}
				text="load more"
			/>
		);

		const button = screen.getByRole('button', { name: /load more/i });
		expect(button).toBeInTheDocument();
	});

	it('should call a function on button click', () => {
		render(
			<Button
				onClick={fn}
				disabled={false}
				text="load more"
			/>
		);

		const button = screen.getByRole('button', { name: /load more/i });
		fireEvent.click(button);

		expect(fn).toHaveBeenCalled();
	});

	it('should be disabled when disable prop is true', () => {
		render(
			<Button
				onClick={fn}
				disabled={true}
				text="load more"
			/>
		);

		const button = screen.getByRole('button', { name: /load more/i });

		expect(button).toBeDisabled();
	});

	it('should match snapshot', () => {
		const { container } = render(
			<Button
				onClick={fn}
				disabled={true}
				text="load more"
			/>
		);

		expect(container.firstChild).toMatchSnapshot();
	});
});
