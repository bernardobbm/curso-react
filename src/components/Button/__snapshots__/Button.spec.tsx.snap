// Vitest Snapshot v1

exports[`Button Component > should match snapshot 1`] = `
<button
  class="button"
  disabled=""
>
  load more
</button>
`;
